packages/common/core-interfaces/src/jsonDeserialized.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { InternalUtilityTypes } from "./exposedUtilityTypes.js";
+
+/**
+ * Options for {@link JsonDeserialized}.
+ *
+ * @beta
+ */
+export interface JsonDeserializedOptions {
+	/**
+	 * A type that is managed by custom deserialization logic (beyond JSON.parse).
+	 *
+	 * The default value is `never`.
+	 */
+	Replaced: unknown;
+}
+
+/**
+ * Produces a type that results from a type `T` serialized and deserialized
+ * through JSON.
+ *
+ * @typeParam T - The type that was serialized.
+ *
+ * @remarks
+ * When used as a filter to inferred generic `T`, a compile-time error can be
+ * produced trying to assign `JsonDeserialized<T>` to `T`.
+ *
+ * Deserialized JSON never contains `undefined`, `symbol`, or function values.
+ * Object properties with values of those types are absent. So properties
+ * become optional (when there are other supported types in union) or are
+ * removed (when nothing else in union is supported).
+ * In an array, such values are replaced with `null`.
```

Comment:
When using tools like git which are line based, personally I prefer more semantic line breaks over arbitrary ones mainly
Starting sentences on new lines. Also, I don't think starting a sentence with a coordinating conjunction like "so" like that is correct. either than period should become a comma, or we can use a colon here (which I think makes the most sense):

```suggestion
 * Deserialized JSON never contains `undefined`, `symbol`, or function values.
 * Object properties which had those values before encoding are omitted by JSON.stringify:
 * Therefor in this type properties become optional (when there are other supported types in union)
 * or are removed (when nothing else in union is supported).
 * In an array, such values are replaced with `null`.
```